File: 面试真题/acm_template.py
"""
常用运行指令: python3 acm_template.py
快速测试指令 (从文件读入): python3 acm_template.py < input.txt
"""
import sys
import collections
import heapq
import math
from typing import List, Optional

class Solution:
    def solve(self):
        """
        在这里实现你的核心逻辑
        """
        # TODO: 补充具体的解题代码
        pass

def main():
    # ==========================================
    # 模式1：sys.stdin.read().split() 
    # 优势：一次性把所有的输入全部读进来并按空格/换行切成列表，
    # 提取数字特别方便，适合大部分纯数字或无空格字符串题
    # ==========================================
    """
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    # 用一个迭代器方便提取数据
    it = iter(input_data)
    
    # 举例：读取 t 组数据
    try:
        t = int(next(it))
        for _ in range(t):
            # n = int(next(it))
            # arr = [int(next(it)) for _ in range(n)]
            
            sol = Solution()
            sol.solve()
    except StopIteration:
        pass
    """

    # ==========================================
    # 模式2：sys.stdin 按行读取 (默认打开)
    # 优势：适合一行一行有严格结构的数据，或者带有空格的字符串题
    # ==========================================
    for line in sys.stdin:
        # 去掉换行符和首尾空格
        line = line.strip()
        if not line:
            continue
        
        # TODO: 在这里解析当前行数据，然后传递给 solution
        # 比如：a, b = map(int, line.split())
        
        sol = Solution()
        sol.solve()

if __name__ == "__main__":
    # 为了避免递归深度超过限制，有些题需要解开这个注释
    # sys.setrecursionlimit(200000)
    main()

File: 面试真题/acm_template.cpp
/*
 * 常用编译指令: g++ -O2 -std=c++17 acm_template.cpp -o acm_template
 * 运行指令: ./acm_template
 * 快速测试指令 (从文件读入): ./acm_template < input.txt
 */
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <unordered_map>
#include <map>
#include <set>
#include <queue>
#include <stack>
#include <cmath>

using namespace std;

class Solution {
public:
    // 在这里实现你的核心逻辑
    void solve() {
        // TODO: 补充具体的解题代码
        
    }
};

int main() {
    // 优化标准I/O流性能，ACM模式下必备
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    // ==========================================
    // 模式1：多组测试数据，第一行给定了组数 t
    // ==========================================
    /*
    int t;
    if (cin >> t) {
        while (t--) {
            Solution sol;
            sol.solve();
        }
    }
    */

    // ==========================================
    // 模式2：持续读取直到 EOF (遇到文件尾或Ctrl+D)
    // ==========================================
    /*
    int n;
    while (cin >> n) {
        // 读取其它参数
        Solution sol;
        sol.solve();
    }
    */

    // ==========================================
    // 模式3：单组测试数据（默认打开）
    // ==========================================
    Solution sol;
    sol.solve();

    return 0;
}
